libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1318,11 +1318,41 @@ bool TypeChecker::visit(Conditional const& _conditional)
 	return false;
 }
 
+void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const& _expression)
+{
+	if (auto const* tupleExpression = dynamic_cast<TupleExpression const*>(&_expression))
+	{
+		auto const* tupleType = dynamic_cast<TupleType const*>(&_type);
+		auto const& types = tupleType ? tupleType->components() : vector<TypePointer> { _type.shared_from_this() };
```

Comment:
The lifetime of the temporary does not extend beyond this line, you have to use `vector<TypePointer>` for the type.

Reply:
Turns out this is not true, references extend the lifetime of temporaries bound to them.